lib/src/components/NxDropdown/__tests__/NxDropdown.test.tsx

```diff
@@ -5,357 +5,301 @@
   });
 
   it('does not call onToggleCollapse if ESC is pressed within the component when the component is disabled',
       function() {
-        const onToggleCollapse = jest.fn(),
-            component = getMountedComponent({ onToggleCollapse, isOpen: true, disabled: true });
+        const onToggleCollapse = jest.fn();
+        const component = renderEl({ onToggleCollapse, isOpen: true, disabled: true })!;
 
         expect(onToggleCollapse).not.toHaveBeenCalled();
-
-        component.simulate('keyDown', { key: 'Escape' });
+        fireEvent.keyDown(component, { key: 'Escape' });
         expect(onToggleCollapse).not.toHaveBeenCalled();
       }
   );
 
   it('does not call onToggleCollapse if ESC is pressed within the component and onCloseKeyDown preventsDefault',
       function() {
-        const onToggleCollapse = jest.fn(),
-            component = getMountedComponent({
-              onToggleCollapse,
-              isOpen: true,
-              onCloseKeyDown: e => e.preventDefault()
-            }, { attachTo: container });
-
-        act(() => {
-          component.find('button.nx-dropdown__toggle').getDOMNode()
-              .dispatchEvent(new KeyboardEvent('keydown', { key: 'Escape', bubbles: true }));
-        });
-        component.update();
+        const onToggleCollapse = jest.fn();
+
+        const props: Partial<Props> = {
+          onToggleCollapse,
+          isOpen: true,
+          onCloseKeyDown: e => e.preventDefault()
+        };
+
+        const { rerender } = quickRender(props);
+
+        const button = screen.getByRole('button');
+        fireEvent.keyDown(button, { key: 'Escape', bubbles: true });
+
+        rerender(<NxDropdown {...minimalProps} {...props} />);
+
         expect(onToggleCollapse).not.toHaveBeenCalled();
       }
   );
 
-  it('does not call onToggleCollapse if a click happens when onCloseClick preventsDefault', function() {
-    const onToggleCollapse = jest.fn(),
-        component = getMountedComponent({
-          children: <button className="nx-dropdown-button">Foo</button>,
-          onToggleCollapse,
-          isOpen: true,
-          onCloseClick: e => e.preventDefault()
-        }, { attachTo: container });
+  it('does not call onToggleCollapse if a click happens when onCloseClick preventsDefault', async function() {
+    const user = userEvent.setup();
+    const onToggleCollapse = jest.fn();
+
+    const props: Partial<Props> = {
+      children: <button className="nx-dropdown-button" data-testid="dropdown-button">Foo</button>,
+      onToggleCollapse,
+      isOpen: true,
+      onCloseClick: e => e.preventDefault()
+    };
+
+    const { rerender } = quickRender(props);
 
     expect(onToggleCollapse).not.toHaveBeenCalled();
 
-    act(() => {
-      document.dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    component!.update();
+    await user.click(document.body);
+    rerender(<NxDropdown {...minimalProps} {...props} />);
```

Comment:
These rerenders also seem unnecessary. Are you under the impression that this is necessary to potentially call `onToggleCollapse`? I don't think it is

Reply:
> These rerenders also seem unnecessary. Are you under the impression that this is necessary to potentially call `onToggleCollapse`? I don't think it is. It's not like in enzyme where we had to randomly call `component.update()` for reasons we never understood

Thanks for that clarification: I've removed it here: 875cc55fa5870d28a090056472f32c6d5c2f90bb